java/src/main/java/com/bazaarvoice/gumshoe/Gumshoe.java

```diff
@@ -46,10 +46,10 @@ public static Gumshoe get() {
         }
 
         Gumshoe instance = instances.get();
-        if (instances.get() == null) {
+        if (instance == null) {
             EventFactory eventFactory = new EventFactory();
             EventHandler eventHandler = new EventHandler(getConfiguration());
-            instance = new Gumshoe(eventFactory, eventHandler);
+            instance = new Gumshoe(eventFactory, eventHandler, configuration.getMaxContextStackDepth());
```

Comment:
The short answer is that eventing is a cross-cutting concern and the events may need to be related together as some sort of unit of work (i.e. an initial context) that spans multiple classes that work together, but are decoupled by either a controlling class, DI or whatever.  Data for events may need to be shared between these classes, but we want them to still be decoupled.  This design allows for that.

If that still does not make sense because its too abstract, lets discuss in person and I can make it clear with a concrete example.  It may help to look at the demo project in this repo which is a working concrete example.  Look at the main ```Application``` code and the collaborating ```AutoSalesAggregator``` class.  In reality, a business process that we want to event may span many different classes, even different applications.